apps/cowswap-frontend/src/modules/permit/utils/appDataContainsHooks.ts

```diff
@@ -0,0 +1,16 @@
+import { AppDataRootSchema } from '../../appData/types'
+
+export function appDataContainsHooks(fullAppData: string | undefined): boolean {
+  if (!fullAppData) {
+    return false
+  }
+
+  try {
+    const appData = JSON.parse(fullAppData) as AppDataRootSchema
+
+    return !!appData.metadata?.hooks
+  } catch (e) {
+    // Does not match the schema, use a regex
+    return /"hooks"/.test(fullAppData)
```

Comment:
mm, a bit weird check... 
so if the appData has any field with `"hooks"`, then you return true?

what case are you trying to cover? if its us who generated the hook, then it won't be an issue

Reply:
And the regex is just a precaution in case for some reason we fail to parse the appData.

Reply:
But what i mean, isn't it covered by `const appData = JSON.parse(fullAppData) as AppDataRootSchema`, why would it have some other JSON structure? testing for the presence of "hooks" feels fragile

Reply:
Fair, the regex is an overkill.
If the data doesn't match the expected format it'll probably not be executable anyway.
I'll remove this bit.